### PR TITLE
Add github-actions and woodpecker

### PR DESCRIPTION
**Pull request by phreed**
PR Checklist:

- [x] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] put any other relevant information below

Corrected some minor typographical errors.
The parameters for github_actions and woodpecker may be incomplete.

This is related to https://github.com/conda-forge/conda-forge.github.io/discussions/1746

### CODE REVIEW

**Comment by phreed** [comment]
Error when running the `make html` .
```
/mnt/d/ip/conda-forge.github.io/src/maintainer/infrastructure.rst:: ERROR: Anonymous hyperlink mismatch: 1 references but 0 targets.
See "backrefs" attribute for IDs.
```

File: docs/_sources/maintainer/infrastructure.rst.txt [hunk 1]
```diff
@@ -272,8 +272,8 @@ here `<https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/
 GitHub Actions
 --------------
 
-We use GitHub actions to rerender feedstocks and run our pull request automerge service. We do not currently support builds on 
-GitHub Actions. 
+We use GitHub actions to rerender feedstocks and run our pull request automerge service. We do not currently support builds on
+GitHub Actions.
```
Comment (jakirkham) [inline docs/_sources/maintainer/infrastructure.rst.txt]: Can you please revert these? The files in this directory are autogenerated

File: src/maintainer/conda_forge_yml.rst [hunk 2]
```diff
@@ -270,9 +272,20 @@ defaults are as follows:
       repo_name: ""
       # branch name to execute on
       branch_name: master
-      # branch name to use for rerender+webservices github actions and
-      # conda-forge-ci-setup-feedstock references
-      tooling_branch_name: master
```
Comment (jakirkham) [inline src/maintainer/conda_forge_yml.rst:275]: Why is this being deleted?

**Comment by jakirkham** [comment]
Know I've been commenting above, but is this ready for review? If so, may be worth taking out of draft mode to get more feedback

File: src/maintainer/conda_forge_yml.rst [hunk 3]
```diff
@@ -537,3 +555,22 @@ Currently only:
...
+
+.. code-block:: yaml
+
+    woodpecker:
+      # Flag for uploading to anaconda. (common setting)
+      upload_packages: True
+      # woodpecker specific settings
+      yum_specific_setup:
+      build_setup:
+
```
Comment (phreed) [inline src/maintainer/conda_forge_yml.rst:576]: I am looking for some guidance on `woodpecker`.
I suspect it is similar to `dune.io` but that seems to have either been removed or never provided.

**Comment by phreed** [comment]
* https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/configure_feedstock.py#L1243
* https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/configure_feedstock.py#L1767-L1772

File: src/maintainer/conda_forge_yml.rst [hunk 4]
```diff
@@ -274,6 +276,20 @@ defaults are as follows:
...
+This dictates the behavior of the Github Actions CI service.
+It is a mapping for Github-Action-specific configuration options.
+For example:
+
+.. code-block:: yaml
+
+    github_actions:
+      # Flag for uploading to anaconda.
+      upload_packages: True
+
```
[the remark this answers is not in the document]
Comment (phreed) [reply src/maintainer/conda_forge_yml.rst]: https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/configure_feedstock.py#L1767-L1772
https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/configure_feedstock.py#L1243
https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/configure_feedstock.py#L651
https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/templates/github-actions.tmpl

File: src/maintainer/conda_forge_yml.rst (hunk 3 above)
Comment (phreed) [reply src/maintainer/conda_forge_yml.rst]: * https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/templates/woodpecker.yml.tmpl
* https://github.com/conda-forge/conda-smithy/blob/bb2bb5aae536acd4867f63584ec0dc5a466db164/conda_smithy/configure_feedstock.py
These did not offer much illumination.

**Comment by phreed** [comment]
> Know I've been commenting above, but is this ready for review? If so, may be worth taking out of draft mode to get more feedback

I changed it to draft because I found information about the parameters for `github_actions` and `woodpecker`.
Now that they have been updated I think it is ready for review.

**Comment by jakirkham** [comment]
@conda-forge/core, any thoughts on this one? 🙂

File: src/maintainer/conda_forge_yml.rst [hunk 5]
```diff
@@ -133,7 +135,7 @@ long-term support (LTS) branch for a package.
 build_platform
 --------------
 This is a mapping from the target platform to the build platform for the package
-to be built. e.g. the following builds a ``osx-64`` package on the ``linux-64``
+to be built. e.g. the following builds a ``osx_64`` package on the ``linux_64``
```
Comment (jaimergp) [inline src/maintainer/conda_forge_yml.rst:138]: Why the change to underscores? I would swear `build_platform` uses dashes, like every other conda subdir string.

File: src/maintainer/conda_forge_yml.rst [hunk 6]
```diff
@@ -226,7 +228,7 @@ modified.  Tools like conda-smithy may modify this, as needed.  It has a single
 
 .. code-block:: yaml
 
-    appveyor:
+    circle:
       secure:
         BINSTAR_TOKEN: <some big hash>
```
Comment (jaimergp) [inline src/maintainer/conda_forge_yml.rst:233]: I think we are removing this block in other PR so this change might be moot.

See https://github.com/conda-forge/conda-forge.github.io/pull/1793

File: src/maintainer/infrastructure.rst [hunk 7]
```diff
@@ -285,8 +285,14 @@ here `<https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/
 GitHub Actions
 --------------
 
-We use GitHub actions to rerender feedstocks and also run our pull request automerge service. We do not currently support builds on
-GitHub Actions.
+We use GitHub actions to rerender feedstocks and also run our pull request automerge service.
+We also support builds on GitHub Actions.
```
Comment (jaimergp) [inline src/maintainer/infrastructure.rst:289]: This is not entirely true. There's support for GitHub Actions workflows on conda-forge as long as they do not use Github-hosted runners. However, users can use conda-smithy to populate a feedstock (outside conda-forge!) that does use Github Actions with Github-hosted runners. The reasoning is that conda-forge is too big for GH-hosted runners on our org, and we would easily consume all the runners that are used to run other infrastructure services.

File: src/maintainer/infrastructure.rst [hunk 8]
```diff
@@ -285,8 +285,14 @@ here `<https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/
...
+
+Woodpecker
+----------
+
+We support `Woodpecker <https://github.com/woodpecker-ci/woodpecker>`__ for Linux ARMv8 builds.
```
Comment (jaimergp) [inline src/maintainer/infrastructure.rst:294]: Woodpecker is only the project that orchestrates CI workflows, I think. There's no hosted service running on their end; that needs to be provided by the user (us). I think we used Woodpecker at some point to run jobs on a donated machine, but I am not sure this is still running.

File: src/maintainer/conda_forge_yml.rst [hunk 9]
```diff
@@ -537,3 +560,19 @@ Currently only:
 
     win:
       enabled: False
+
+.. _woodpecker:
+
+woodpecker
+----------
+This dictates the behavior of the woodpecker CI service (a fork of Dune).
```
Comment (jaimergp) [inline src/maintainer/conda_forge_yml.rst:568]:
```suggestion
This dictates the behavior of the woodpecker CI service (a fork of Drone).
```

Typo?

**Review by jaimergp — COMMENTED** [review]
Some comments from my limited understanding! Somebody else from core should double check what I said it's true 😬

**Comment by jaimergp** [comment]
`github_actions` is now documented via other PRs, and Woodpecker never made it so I'm going to close this PR. Thanks for the effort!
